### PR TITLE
Fix/security issues

### LINKED ISSUE #18854: High/medium nsp vulnerabilities for dependency ws and hoek
Similar to vulnerability issues in the past ([here](https://github.com/facebook/react-native/issues/18394) and [here](https://github.com/facebook/react-native/issues/7643)), we have 2 outstanding nsp vulnerabilities.
Running `nsp check` produces the following:
```
(+) 2 vulnerabilities found
┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Denial of Service                                                  │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ ws                                                                 │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 7.5 (High)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 2.3.1                                                              │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <1.1.5 || >=2.0.0 <3.3.1                                           │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ >= 1.1.5 <2.0.0 || >=3.3.1                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ my-app@0.1.0 > react-native@0.55.2 > react-devtools-core@3.1.0     │
│            │ > ws@2.3.1                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/550                             │
└────────────┴────────────────────────────────────────────────────────────────────┘

┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Prototype pollution attack                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ hoek                                                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 4 (Medium)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 2.16.3                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                        │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ my-app@0.1.0 > react-native@0.55.2 > metro@0.30.2 >                │
│            │ jest-haste-map@22.4.2 > sane@2.5.0 > fsevents@1.1.3 >              │
│            │ node-pre-gyp@0.6.39 > hawk@3.1.3 > hoek@2.16.3                     │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/566                             │
└────────────┴────────────────────────────────────────────────────────────────────┘
```

## Environment
```
Environment:
  OS: macOS Sierra 10.12.6
  Node: 8.9.4
  Yarn: 1.3.2
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  react: ^16.0.0 => 16.2.0
  react-native: ^0.55.0 => 0.55.2
```

## Steps to Reproduce
```
npm install -g nsp
nsp check
```

## Expected Behavior
```
(+) 0 vulnerabilities found
```

## Actual Behavior
```
(+) 2 vulnerabilities found
```
( Details above )

These are getting picked in the whitesource vulnerability audit, resolving these would be a huge confidence booster.

### PR DESCRIPTION
**Pull request by byCedric**
This updates both [`react-devtools-core`](https://github.com/facebook/react-devtools/issues/941) and [`plist`](https://github.com/TooTallNate/plist.js/commit/04c8ee7646d4abcfe13b3fdc3037b75047069ead) to include security fixes, reported by `npm audit`. 

Fixes #18854 

## Test Plan

_Only dependencies are updated with security patches, no change in actual code._

## Related PRs

_Only dependencies are updated with security patches, no change in actual code._

## Release Notes

[GENERAL] [ENHANCEMENT] [react-devtools-core] - Bump to `3.2.2` for a security fix for `ws`
[GENERAL] [ENHANCEMENT] [plist] - Bump to `3.0.0` for a security fix for `xmlbuilder`

### CODE REVIEW

**Comment by react-linter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :lock: package.json - <i>Changes were made to package.json. This will require a manual import by a Facebook employee.</i>
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by janicduplessis** [comment]
cc @hramos for landing this

**Comment by byCedric** [comment]
Regarding the comment of @hramos in the other thread, mentioned by @rajivshah3 (sorry dude, didn't see your PR):

> Before I merge this, can you tell me why you think this does not need to be tested? Where is plist used? I am hesitant to land a PR that bumps a package by two major versions with no testing done.

It's a fair point you raised here. By looking at the [`History.md`](https://github.com/TooTallNate/plist.js/blob/master/History.md), I think what @janicduplessis did already, you can see that there aren't big "breaking changes" as you would expect when looking at semver. I'm not sure why they bumped the versions from `1.2.0` -> `2.0.0` -> `2.0.1` -> `2.1.0` -> `3.0.0`. Personally, I'm not sure how the library is being maintained right now but I think there are some improvements to make... Especially when looking at the reason of [`2.0.1`](https://github.com/TooTallNate/plist.js/blob/master/History.md#201--2016-08-16)... The biggest changes I can find are the following.

#### `1.2.0` => `2.0.0`
> remove deprecated functions ([90deef5](https://github.com/TooTallNate/plist.js/commit/90deef5d431c17b4b76c5ad51817d54bd0de555b))

These methods were deprecated already starting from `1.0.0`, although that what the history file is showing to me. I don't think this affects the current code base because it was already using `1.2.0`.

#### `2.0.1` => `2.1.0`
> update license (extend copyright term) ([e4f0666](https://github.com/TooTallNate/plist.js/commit/e4f06669bb51d2e65654df7c39aab52bc3bf4e8a))

Well, with the [Facebook's BSD+Patents "gate" freshly in mind](https://hackernoon.com/facebooks-bsd-patents-license-and-how-it-affects-you-66088e052845) I started to get more careful with licences. That's why I think this is the bigest change here that might deserve a little love and time.

#### `2.1.0` => `3.0.0`
> update module deps, remove support for node < 6 ([c962bfe](https://github.com/TooTallNate/plist.js/commit/c962bfe1ea04dd87966250253d33738bceecab61))

This might be a big change. Luckily for us, React Native is limited to [`node@>=8`](https://github.com/facebook/react-native/blob/master/package.json#L11) as it should be. So this doesn't affect the code again.

So long story short, It's a fair point. But I don't think the maintainers from `plist` are using semver strictly. And when you look at the actual changes, it doesn't look like it should affect the code much. But then again, this is my first contribution to React Native itself 😬 

Hope it helps!
❤️ Cedric

**Comment by byCedric** [comment]
Is there any news on when we can expect this to be merged? We really want to set node security issue check to blocking again in our pipeline 😬

**Comment by hramos** [comment]
Working on it. Any PR that touched `package.json` requires manual effort to land, hang tight.

**Comment by jeiemgi** [comment]
Hi @hramos, or anybody who could help me.

I'm sorry for this dumb question, but I have RN 0.55.4 version and I don't know how to apply this fix.

- I must upgrade RN version ?,
- To which version ?,
- Is there a way I can apply this fix, and don't upgrade RN version?

**Comment by hramos** [comment]
This fix was merged several months after 0.55.4 was released, so you will need to upgrade.

If you look at the commit that closed this PR, you'll see the change is available in any of the following releases:

- v0.57.0-rc.0
- v0.56.0
- v0.56.0-rc.5
- v0.56.0-rc.4
- v0.56.0-rc.3
- v0.56.0-rc.2
- v0.56.0-rc.1
- v0.56.0-rc
- latest (master)

**Comment by hramos** [comment]
If you're only interested in this fix, you should be able to apply this patch over the `0.55-stable` branch in order to roll out your own custom 0.55 release. GitHub provides a handy way of getting a patch from a PR, just add .patch to the URL like so https://github.com/facebook/react-native/pull/19373.patch

**Comment by jeiemgi** [comment]
Thank you, and thank you for the patience.
